pyproject.toml

```diff
@@ -1,11 +1,143 @@
 [build-system]
-# Minimum requirements for the build system to execute.
-requires = ["setuptools>=61.0.0", "setuptools_scm>=7.0.1"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.0.0", "hatch-vcs"]
+build-backend = "hatchling.build"
 
-[tool.setuptools_scm]
-write_to = "src/pyhf/_version.py"
-local_scheme = "no-local-version"
+[project]
+name = "pyhf"
+dynamic = ["version"]
+description = "pure-Python HistFactory implementation with tensors and autodiff"
+readme = "README.rst"
+license = { text = "Apache-2.0" }  # SPDX short identifier
+requires-python = ">=3.8"
+authors = [
+    { name = "Lukas Heinrich", email = "lukas.heinrich@cern.ch" },
+    { name = "Matthew Feickert", email = "matthew.feickert@cern.ch" },
+    { name = "Giordon Stark", email = "gstark@cern.ch" },
```

Comment:
This information also ends up in the wheel under `LICENSE-FILE` proposed metadata entry (PEP 639), due to AUTHORS existing. That's probably fine.

https://github.com/pypa/hatch/blob/8a9ba2091c2ccb49021de5d44595f535c36331c9/backend/src/hatchling/metadata/core.py#L712

Reply:
Yeah I think it is okay to have this duplicated. I really only cared about getting `AUTHORS` in the sdist.

Reply:
Agreed - another "I noticed this whilst here"